Protein localization phenotype design patterns

rays22 opened the pull request:
This commit intends to add three new design patterns related to protein localization phenotypes.
If applied, this commit will fix #753.

matentzn reviewed (approved):
Good job @rays22! They look very reasonable to me, nicely dione. They need to be reviewed by @dosumis and @pfey03 as well.

Just two things for Pull request:
1. Always name the ticket the pull request is solving (if it is fixing it, write `fixes #123`
2. Always provide a small two line summary about that the PR does, and if the review should pay special attention to something you are particularly unsure of.

dosumis commented:
@mah11 - 'increased rate' or 'increased efficacy' - rate feels a bit too narrow here - although there's still work to do to decide the relationship of these 2 terms in Uberon.

dosumis reviewed (commented):
Looks good.  Minor issue with class naming.  - see comment.  Also want to wait on feedback from Midori about which PATO terms to use before merging.

matentzn commented on src/patterns/dosdp-dev/abnormalProteinLocalizationInCellularComponent.yaml:
```diff
@@ -0,0 +1,42 @@
+  abnormal: PATO:0000460
+  rate: PATO:0000161
+  protein localization: GO:0008104
+  sub-cellular location: GO:0005575
```
@cmungall recomments to move away from structural names in patterns (the kind of thing the filler is) to "functional" name - the kind of thing the filler is achieving for this pattern. It was my recommendation to do that here. Do you disagree?

dosumis replied:
Makes sense for vars.  To what extent should we be changing the names used to refer to referenced classes within the pattern - as here?

dosumis replied:
In this case, if you want to make clear that location should be subcellular, shouldn't we just indicate that in the var name and keep the range specification faithful to the name of the term used?  Not a big deal, but I worry that this approach might cause some confusion in future.  It's possible to use misleading names for referenced terms.

matentzn replied:
No you are probably right. @rays22 can you use "cellular component" in the classes section please?

mah11 replied:
> @mah11 - 'increased rate' or 'increased efficacy' 

In FYPO we have use cases for both. For the plain 'increased protein localization to $place' pattern I'm inclined towards 'increased efficacy', on the grounds that I interpret 'rate' as "how fast" or "how many iterations per unit time".

We have not asserted class-superclass links between any 'increased|decreased efficacy' and 'increased|decreased rate' terms (generally, not just for protein localization processes). Same goes for changed occurrence - "sibling" of changed rate.

matentzn replied:
Thank you @mah11 this is kinda what I though as well. Efficacy seems more generally useful at the moment. @pfey03, can you also check whether increased efficacy (~increased quality of output) would work fo you rather than increased rate? Or shall we create both?

dosumis replied:
>  For the plain 'increased protein localization to $place' pattern I'm inclined towards 'increased efficacy', on the grounds that I interpret 'rate' as "how fast" or "how many iterations per unit time".

Agreed.  In reality the observation is most likely to be increased accumulation at some location.  Mechanism could be change in rate active localization, active degredation, change in stability etc so a more general term makes sense. 

@rays22  can you chance the PATO term and the name to fit?


> We have not asserted class-superclass links between any 'increased|decreased efficacy' and 'increased|decreased rate' terms (generally, not just for protein localization processes). Same goes for changed occurrence - "sibling" of changed rate.

Yeh - we need a general solution for this.  I've been promising one for too long.  Would you be happy for us to move PATO rate and occurence terms under efficacy terms?

matentzn replied:
@dosumis Such a big questions should better be asked on the PATO tracker directly, and then circulated to all stakeholders.. @rays22 can you make a ticket to ask 

> Would you be happy for us to move PATO rate and occurence terms under efficacy terms?

Linking this PR as well? It seems a bit.. risky as a general move, but lets see maybe its practical!

mah11 replied:
>  In reality the observation is most likely to be increased accumulation at some location.

Indeed, and with hindsight I sometimes wonder whether we should have made more of the FYPO 'protein localization to x' terms 'protein level at/in x' instead. But not all, because every now and then some clever experimenter really does monitor a localization process in near enough to real time ...

If 'efficacy' is general enough, we get to dodge revamping a _lot_ of terms ;)

> Would you be happy for us to move PATO rate and occurence terms under efficacy terms?

I think so ... at least, nothing leaps out as an obvious problem with doing so.

pfey03 replied:
The abnormal pattern is fine
abnormalProteinLocalizationInCellularComponent
But why adding RateOf for the decreased and increased?. As a localization, even when there is much less or much more in a mutant cell, the rate is never measured. I would prefer it simpler like the abnormal term.

I prefer to all be localization but could live with accumulation for increased.

Also, I need as well **absent:**
abnormallyAbsentProteinLocalizationInCellularComponent

pfey03 replied:
efficacy is definitely better than rate as rate is too specific. efficacy is more general.

matentzn replied:
abnormal is not a quality - it is a modifier of a quality (like size, rate, etc etc). We try to find the most specific yet useful quality in PATO for our patterns - the fallthrough (what you call "abnormal") is to use PATO:quality as the quality in question, but its highly problematic. Increased quality has literally no meaning! So we need to specify - what "quality" of the process is affected? Its rate? Its ability to produce output? This pattern is not the same as saying "there is more protein" - that would be a different pattern. This is about saying that the ability of the process of protein localisation is somehow more.. But from what you are saying, efficacy is the way to go.

pfey03 replied:
Thanks @matentzn

pfey03 replied:
What about **absent** pattern? Or did I miss it?

matentzn replied:
@rays22 please add the corresponding absent pattern, modelled after the absent biological process pattern

rays22 commented:
I have tried to address the issues raised in the comments. In the latest commit I have
- added one more new design pattern related phenotypes where protein localization is abnormally absent,
- changed the classes section using the term `cellular component`,
I have also added contributor attributions.
